desc/io/input_reader.py

```diff
@@ -691,6 +691,13 @@ def write_desc_input(filename, inputs, header=""):
 
         f.write("\n# solver tolerances\n")
         for key in ["ftol", "xtol", "gtol", "nfev"]:
+            at_least_one_is_none = False
+            for inp in inputs:
+                if inp[key] is None:
+                    at_least_one_is_none = True
+            if at_least_one_is_none:
+                break  # don't write if an input tolerance is None
```

Comment:
This won't write any of the stopping criteria if at least one of them is None. Why not only skip the ones that are None? If a value is passed in for one parameter, it seems odd to overwrite it with the default just because some other parameter was not specified.

Reply:
The logic was just getting a bit annoying to do. I can make that change